pkg/nodejs/prepare/application.go

```diff
@@ -99,7 +114,7 @@ http {
        listen 8080;
 
        location /api {
-          proxy_pass http://${PROXY_PASS_HOST}:${PROXY_PASS_PORT};
+          {{if .HasNodeJSApplication}}proxy_pass http://${PROXY_PASS_HOST}:${PROXY_PASS_PORT};{{else}}return 404;{{end}}
```

Comment:
Betyr dette at det ikke er mulig å sette opp en proxy til en annen backend hvis det ikke finnes en NodeJs applikasjon? Var ikke dette en av mulighetene som skulle være tilgjengelig for Webleveransepakke?

Reply:
Godt poeng.....

Jeg ser at vi nok må inn med en konfigvariabel til. Det jeg har tenkt, er at det er enkelte applikasjoner som ikke har backend, og dermed ikke skal konfigureres med noe proxy-pass, men det er jo helt riktig som du sier at det bør være mulig selv om du ikke har en nodejs backend.

Hva tror du om å legge til nytt felt i openshift.json med "configurableProxy": true ? Om det er satt til "true", så impliserer det ingen nodebackend. Hvis den ikke er satt, så har man nodebackend, og ingen mulighet for konfigurertbart proxypass.

Reply:
Burde vi åpne for en mer fleksibel løsning? For eksempel å kunne sende log til nodejs og api til java-backend.
openshift.json
```
"proxy": {
    "/api": {
      "host": "<java-backend>",
      "port": 9000
    },
    "/log": {
      "host": "<nodejs-backend>",
      "port": 9001
    }
}
```
Men hvis vi ønsker å ha kontroll på hvilke endepunk som kan konfigureres med proxy så er din løsning med "configurableProxy" bra.

Reply:
Jeg tenker at proxy-konfigurasjon er runtime-konfig, så den bør ikke ligge i openshift.json. Men jeg ser at det kan være hensiktsmessig å kunne konfigurere flere.

Jeg har ikke lyst til å la dette bli for omfattende nå. Kan vi løse dette i en eventuell fremtidig sak om vi ser behovet kommer?

Reply:
Enig i det, vi går for "configurableProxy" i denne omgang.